dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java

```diff
@@ -0,0 +1,517 @@
+package org.jboss.resteasy.plugins.providers.sse.client;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import javax.ws.rs.ServiceUnavailableException;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response.Status.Family;
+import javax.ws.rs.ext.Providers;
+import javax.ws.rs.sse.InboundSseEvent;
+import javax.ws.rs.sse.SseEventSource;
+
+import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
+import org.jboss.resteasy.client.jaxrs.internal.ClientConfiguration;
+import org.jboss.resteasy.client.jaxrs.internal.ClientInvocation;
+import org.jboss.resteasy.client.jaxrs.internal.ClientResponse;
+import org.jboss.resteasy.plugins.providers.sse.SseConstants;
+import org.jboss.resteasy.plugins.providers.sse.SseEventInputImpl;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+
+public class SseEventSourceImpl implements SseEventSource
+{
+    public static final long RECONNECT_DEFAULT = 500;
+
+   private final WebTarget target;
+
+   //Liberty... removed final to allow change.
+   private long reconnectDelay;
+
+   private final SseEventSourceScheduler sseEventSourceScheduler;
+
+   private enum State {
+      PENDING, OPEN, CLOSED
+   }
+
+   private final AtomicReference<State> state = new AtomicReference<>(State.PENDING);
+
+   private final List<Consumer<InboundSseEvent>> onEventConsumers = new CopyOnWriteArrayList<>();
+
+   private final List<Consumer<Throwable>> onErrorConsumers = new CopyOnWriteArrayList<>();
+
+   private final List<Runnable> onCompleteConsumers = new CopyOnWriteArrayList<>();
+
+   private final boolean alwaysReconnect;
+
+   private volatile ClientResponse response;
+
+   public static class SourceBuilder extends Builder
+   {
+      private WebTarget target = null;
+      private long reconnect = RECONNECT_DEFAULT;
+
+      private String name = null;
+
+      private ScheduledExecutorService executor;
+
+      //tck requires this default behavior
+      //Liberty... setting this to false for now.  Had been true.
+      private boolean alwaysReconnect = false;
```

Comment:
I think this might be nice to be configurable via client properties.

Reply:
agreed